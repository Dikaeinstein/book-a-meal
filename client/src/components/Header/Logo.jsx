import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

const Logo = ({ text, className }) => (
  <NavLink
    to="/"
    activeClassName="active"
  >
    <h1 className={className}>{text}</h1>
  </NavLink>
);

Logo.propTypes = {
  text: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
};

export default Logo;
