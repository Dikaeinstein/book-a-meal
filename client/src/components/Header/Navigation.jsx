import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { removeUser } from '../../helpers/persistUser';
import history from '../../helpers/history';
import { userSignOut } from '../../actions/userActions';

const ConnectedNavigation = ({ urls, signout }) => {
  const handleSignout = (event) => {
    event.preventDefault();
    signout();
    removeUser();
    history.push('/');
  };

  const urlList = urls.map(url => (
    url.link === 'signout'
      ? (
        <li key={url.id}>
          <NavLink
            to={`/${url.link}`}
            onClick={handleSignout}
          >
            {url.name}
          </NavLink>
        </li>
      )
      : (
        <li key={url.id}>
          <NavLink
            to={`/${url.link}`}
          >{url.name}
          </NavLink>
        </li>
      )));

  return (
    <nav className="col-3-4">
      <ul className="nav-container text-center">
        {urlList}
      </ul>
    </nav>
  );
};

const mapStateToProps = state => ({
  urls: state.urls,
});

const mapDispatchToProps = dispatch => ({
  signout() { dispatch(userSignOut()); },
});

ConnectedNavigation.propTypes = {
  urls: PropTypes.arrayOf(PropTypes.object).isRequired,
  signout: PropTypes.func.isRequired,
};

const Navigation = connect(
  mapStateToProps, mapDispatchToProps,
  null, { pure: false },
)(ConnectedNavigation);

export default Navigation;
