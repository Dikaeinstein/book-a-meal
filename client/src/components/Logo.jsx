import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { userSignOut } from '../actions/userActions';

const ConnectedLogo = ({ text, className, goHome }) => {
  const handleClick = () => {
    goHome();
  };
  return (
    <NavLink to="/" onClick={handleClick}>
      <h1 className={className}>{text}</h1>
    </NavLink>
  );
};

ConnectedLogo.propTypes = {
  text: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
  goHome: PropTypes.func.isRequired,
};

const mapDispatchToProps = dispatch => ({
  goHome: () => dispatch(userSignOut()),
});

const Logo = connect(null, mapDispatchToProps)(ConnectedLogo);

export default Logo;
