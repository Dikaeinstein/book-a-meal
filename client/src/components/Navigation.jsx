import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

const ConnectedNavigation = ({ urls }) => {
  const urlList = urls.map(url => (
    <li key={url.id}>
      <a href={url.link}>{url.name}</a>
    </li>
  ));

  return (
    <nav className="col-3-4">
      <ul className="nav-container text-center">
        {urlList}
      </ul>
    </nav>
  );
};

const mapStateToProps = state => ({
  urls: state.urls,
});

ConnectedNavigation.propTypes = {
  urls: PropTypes.arrayOf(PropTypes.object).isRequired,
};

const Navigation = connect(mapStateToProps)(ConnectedNavigation);

export default Navigation;
