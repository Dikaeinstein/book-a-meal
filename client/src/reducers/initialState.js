const initialState = {
  urls: [
    {
      id: 1,
      name: 'Signin',
      link: 'signin',
    },
    {
      id: 2,
      name: 'Signout',
      link: 'signout',
    },
  ],
  menus: {
    isFetching: false,
    data: {
      menu: {
        name: 'Menu for today',
        meals: [],
      },
      menus: [],
    },
    isSet: false,
    isSaving: false,
    fetchError: null,
    saveError: null,
    updateError: null,
    isUpdating: false,
  },
  user: {
    isSubmitting: false,
    loggedIn: false,
    data: {
      id: null,
      name: '',
      email: '',
      role: '',
    },
    error: null,
  },
  orders: {
    isFetching: false,
    isSaving: false,
    isUpdating: false,
    isDeleting: false,
    data: {
      userOrders: [],
      checkedOutOrder: {},
    },
    fetchError: null,
    saveError: null,
    updateError: null,
    deleteError: null,
  },
  meals: {
    isFetching: false,
    isUpdating: false,
    isSaving: false,
    data: [],
    fetchError: null,
    saveError: null,
    updateError: null,
  },
  dashboard: {
    isFetchingTotalOrders: false,
    isFetchingTotalSales: false,
    isUpdating: false,
    isDeleting: false,
    fetchTotalOrdersError: null,
    fetchTotalSalesError: null,
    isFetchingAllOrders: false,
    fetchAllOrdersError: null,
    deleteOrderError: null,
    updateOrderError: null,
    totalAmount: 0,
    totalOrders: 0,
    allOrders: [],
  },
};

export default initialState;
