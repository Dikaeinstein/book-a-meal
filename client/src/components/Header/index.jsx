import React from 'react';
import PropTypes from 'prop-types';
import { ToastContainer } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.min.css';
import Logo from './Logo';
import Navigation from './Navigation';

const Header = ({ className }) => (
  <header className={className}>
    <ToastContainer />
    <Logo
      text="Book-A-Meal"
      className="col-1-4 logo text-left text-dark"
    />
    <div className="toggle-show">&#9776;</div>
    <Navigation />
  </header>
);

Header.propTypes = {
  className: PropTypes.string.isRequired,
};

export default Header;
