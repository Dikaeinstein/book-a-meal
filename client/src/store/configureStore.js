import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import { createLogger } from 'redux-logger';
import throttle from 'lodash.throttle';
import { loadState, saveState } from '../helpers/localStorage';
import initialState from '../reducers/initialState';
import rootReducer from '../reducers';

/**
 * Return redux middleware
 */
const getMiddleware = () => {
  const logger = createLogger();

  const middleware =
    process.env.NODE_ENV !== 'production'
    && process.env.NODE_ENV !== 'test'
      ? applyMiddleware(thunk, logger)
      : applyMiddleware(thunk);

  return middleware;
};

/**
 * Configure the redux store
 *
 * @returns {object} Redux store
 */
const configureStore = () => {
  const persistedState = loadState();

  const state = persistedState
    ? { ...initialState, ...persistedState }
    : initialState;

  const store = createStore(
    rootReducer,
    state,
    compose(
      getMiddleware(),
      window.devToolsExtension ? window.devToolsExtension() : empty => empty,
    ),
  );

  // Throttle the saveState function to be called once every second
  store.subscribe(throttle(() => {
    saveState({ orders: store.getState().orders });
  }, 1000));

  return store;
};

export default configureStore;
