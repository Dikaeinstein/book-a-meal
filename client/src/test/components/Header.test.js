import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import Navigation from '../../components/Navigation';
import configureStore from '../../store/configureStore';
import initialState from '../../reducers/initialState';
import Header from '../../components/Header';
import Logo from '../../components/Logo';
import setup from '../setup';

// Setup tests
setup();

const store = configureStore(initialState);

let wrapper;

describe('<Header />', () => {
  beforeAll(() => {
    /* eslint function-paren-newline: 0 */
    wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <Header className="test" />
        </MemoryRouter>
      </Provider>);
  });
  it('should render a header element', () => {
    expect(wrapper.find('header').length).toEqual(1);
  });
  it('should render a Logo component', () => {
    expect(wrapper.find(Logo).length).toEqual(1);
  });
  it('should render a Navigation component', () => {
    expect(wrapper.find(Navigation).length).toEqual(1);
  });
});
