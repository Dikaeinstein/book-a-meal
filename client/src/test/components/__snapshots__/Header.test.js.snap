// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> shold match snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <MemoryRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "g4fm71",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "g4fm71",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Header
        className="test"
      >
        <header
          className="test"
        >
          <Connect(ConnectedLogo)
            className="col-1-4 logo text-left"
            text="Book-A-Meal"
          >
            <ConnectedLogo
              className="col-1-4 logo text-left"
              goHome={[Function]}
              text="Book-A-Meal"
            >
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                onClick={[Function]}
                to="/"
              >
                <Route
                  path="/"
                >
                  <Link
                    aria-current="true"
                    className="active"
                    onClick={[Function]}
                    replace={false}
                    style={Object {}}
                    to="/"
                  >
                    <a
                      aria-current="true"
                      className="active"
                      href="/"
                      onClick={[Function]}
                      style={Object {}}
                    >
                      <h1
                        className="col-1-4 logo text-left"
                      >
                        Book-A-Meal
                      </h1>
                    </a>
                  </Link>
                </Route>
              </NavLink>
            </ConnectedLogo>
          </Connect(ConnectedLogo)>
          <div
            className="toggle-show"
          >
            â˜°
          </div>
          <Connect(ConnectedNavigation)>
            <ConnectedNavigation
              signout={[Function]}
              urls={
                Array [
                  Object {
                    "id": 1,
                    "link": "signin",
                    "name": "Sign in",
                  },
                  Object {
                    "id": 2,
                    "link": "signup",
                    "name": "Sign up",
                  },
                ]
              }
            >
              <nav
                className="col-3-4"
              >
                <ul
                  className="nav-container text-center"
                >
                  <li
                    key="1"
                  >
                    <NavLink
                      activeClassName="active"
                      ariaCurrent="true"
                      to="/signin"
                    >
                      <Route
                        path="/signin"
                      >
                        <Link
                          aria-current={false}
                          replace={false}
                          to="/signin"
                        >
                          <a
                            aria-current={false}
                            href="/signin"
                            onClick={[Function]}
                          >
                            Sign in
                          </a>
                        </Link>
                      </Route>
                    </NavLink>
                  </li>
                  <li
                    key="2"
                  >
                    <NavLink
                      activeClassName="active"
                      ariaCurrent="true"
                      to="/signup"
                    >
                      <Route
                        path="/signup"
                      >
                        <Link
                          aria-current={false}
                          replace={false}
                          to="/signup"
                        >
                          <a
                            aria-current={false}
                            href="/signup"
                            onClick={[Function]}
                          >
                            Sign up
                          </a>
                        </Link>
                      </Route>
                    </NavLink>
                  </li>
                </ul>
              </nav>
            </ConnectedNavigation>
          </Connect(ConnectedNavigation)>
        </header>
      </Header>
    </Router>
  </MemoryRouter>
</Provider>
`;
